code/game/objects/structures/morgue.dm

```diff
@@ -37,65 +37,87 @@
 		else
 			icon_state = "morgue1"
 
+/obj/structure/morgue/examine(mob/user)
+	..()
+	switch(icon_state)
+		if("morgue2")
+			to_chat(user, "<span class='info'>\The [src]'s light display indicates there is a catatonic corpse inside.</span>")
+		if("morgue3")
+			to_chat(user, "<span class='info'>\The [src]'s light display indicates there are items inside.</span>")
+		if("morgue4")
+			to_chat(user, "<span class='info'>\The [src]'s light display indicates there is an active corpse inside.</span>")
+
 /obj/structure/morgue/ex_act(severity)
 	switch(severity)
 		if(1.0)
-			for(var/atom/movable/A as mob|obj in src)
-				A.loc = src.loc
-				ex_act(severity)
+			for(var/atom/movable/A in src)
+				A.forceMove(src.loc)
+				A.ex_act(severity)
 			qdel(src)
 			return
 		if(2.0)
-			if (prob(50))
-				for(var/atom/movable/A as mob|obj in src)
-					A.loc = src.loc
-					ex_act(severity)
+			if(prob(50))
```

Comment:
I touched the entire fuckin rest of the proc, if that's not good enough then so help me